addons/purchase/models/purchase.py

```diff
@@ -563,21 +563,36 @@ def _inverse_qty_received(self):
             else:
                 line.qty_received_manual = 0.0
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        for values in vals_list:
+            if values.get('display_type', self.default_get(['display_type'])['display_type']):
+                values.update(product_id=False, price_unit=0, product_uom_qty=0, product_uom=False, date_planned=False)
+
+            values.update(self._prepare_add_missing_fields(values))
+
+        lines = super().create(vals_list)
+        for line in lines:
+            if line.product_id and line.order_id.state == 'purchase':
+                msg = _("Extra line with %s ") % (line.product_id.display_name,)
+                line.order_id.message_post(body=msg)
+            if line.order_id.date_planned:
+                line.date_planned = line.order_id.date_planned
+
+        return lines
+
     @api.model
-    def create(self, values):
-        if values.get('display_type', self.default_get(['display_type'])['display_type']):
-            values.update(product_id=False, price_unit=0, product_uom_qty=0, product_uom=False, date_planned=False)
-
-        order_id = values.get('order_id')
-        if 'date_planned' not in values:
-            order = self.env['purchase.order'].browse(order_id)
-            if order.date_planned:
-                values['date_planned'] = order.date_planned
```

Comment:
your change is not strictly equivalent to the previous code

here we set the line's 'date_planned' to the order's one only if it wasn't passed explicitely
here after your change the line's date planned is always set to the order's one

Reply:
Hello @sle-odoo 
changes done.